Docs/source/glossary.rst

```diff
@@ -0,0 +1,232 @@
+
+  MPPT 
+    A maximum power point tracker (MPPT) monitors the power output of a solar array and attempts to maximize that power by adjusting a load (resistance) on that power line. 
+
+  Mutex 
+    A signal that is used to indicate that a thread is using a resource. In multithreaded programming, race conditions become issues. To solve these, a flag is raised each time a resource needs to be accessed, and is unset whenever that resource becomes free for other threads to use. These are typically managed by an operating system or a mediator. 
+
+  Negative Logic 
+    Logical 1/true is a low voltage/off and a logic 0/false is a high voltage/on. 
+
+  NGM 
+    New Generation Motors is a company that makes highly efficient motors. We have used their motors in the past, and often use "the NGM" to refer to the motor that we have in the lab. 
+
+  Open Drain/Collector 
+    A configuration of a pin on a microcontroller where typically logical '0' corresponds to '0V' while logical '1' corresponds to a high impedance state (Hi-Z). This is often used to allow multiple devices to share a single wire for communication. This stands in opposition to a "push-pull" configuration. 
+
+  Op-Amp
+    An operational amplifier is a circuit element that performs mathematical operations on input voltages. It is most commonly used to increase these inputs by a certain magnitude to make them easier to read by the microcontroller.
+
+  Phase Locked Loop
+    The PLL generates an output signal whose phase is related to the input signal. It is used to increase the clock speed of the STM microcontroller.
+
+  Positive Logic 
+    Logical 1/true is a high voltage/on and a logic 0/false is a low voltage/false. 
+
+  Printed Circuit Board
+    PCB's are copper circuits printed onto non-conductive boards that electrically connect our electronic
+    components together and allow us hardware to run our programs on.
+
+  Pull-Down Resistor 
+    A resistor to gnd that pulls a wire's voltage to logical '0' when it disconnects from some other source. 
+
+  Pull-Up Resistor 
+    A resistor to Vcc that pulls a wire's voltage to logical '1' when it disconnects from some other source. 
+
+  Pulse Width Modulation
+    A method of controlling the power usage of a peripheral by quickly turning it on and off and controlling how long it stays on for. Used to control the speed of the fans.
+
+  Push-Pull 
+    A configuration of a pin on a microcontroller where typically logical '0' corresponds to 0V, which logical '1' corresponds to the operating voltage (Vcc). Using this configuration, there can only be one source on a wire that drives communication/control. It stands in opposition to an "open drain" configuration. 
+
+  Queue 
+    A First-In-First-Out (FIFO) structure, where typically supported operations are to insert elements (queue/push), take elements out (dequeue/pop), and view the front element (peek). Elements are removed in the same order that they were entered. 
```

Comment:
I'm curious as to what this is used for in the car. It may not be crucial, but when PWM is described it includes its use in the fans, which I thought was helpful. It helped to create the bigger picture of  how this stuff culminates to create the car. The stack description gives me the same thought: where in the car (aside from user and supervisor stacks for subroutines and interrupts) do we use stacks in.

Reply:
Added more information for those two.